### ISSUE TITLE
`mvn install` fails on multi-module maven project with more than one itest

### ISSUE THREAD

**Issue by mrulli** [bndtools/bnd#1843]
I have a maven project with two modules containing one integration test each. itests are based on bnd-testing-maven-plugin.

* If I `mvn install` from the modules folders, each itest completes successfully.
* If I `mvn install`from the parent pom folder, the second itest fails

I tried to setup [a maven project](https://github.com/mrulli/com.flairbit.examples.bndmultipleitest) to reproduce the issue. 

Sorry for the tests: they are quite nonsensical but I did not have time to think about something better. Anyway this is not important to reproduce the issue: also an empty test would do.

**Comment by timothyjward** [comment]
I have added tests to the `bnd-testing-maven-plugin` to confirm that this isn't an issue on our end. Having tried the tests in your example repository, the first exception I saw was a `NullPointerException` from deep inside Equinox. I switched to using Felix, and the test progressed much further, but still encountered an error in setup.

The exception gets pretty garbled, but the following is visible:

    Caused by: java.lang.ClassNotFoundException: com.sun.xml.bind.DatatypeConverterImpl not found by org.apache.karaf.features.core [24]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1574)
	at org.apache.felix.framework.BundleWiringImpl.access$400(BundleWiringImpl.java:79)
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2018)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 43 more
     found by org.apache.karaf.features.core [24]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1574)
	at org.apache.felix.framework.BundleWiringImpl.access$400(BundleWiringImpl.java:79)
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2018)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 43 more

At this point I'm pretty confident that the plugin is launching things correctly, and that there's something unpleasant going on in the setup of the test.

**Comment by mrulli** [comment]
Thank you for your time and your answer.
Well, I'm quite baffled... Why does the problem only emerge when one runs the tests in the same maven build and the same tests go just fine when you run them separately? Could a bug in the test setup generate entanglement between different tests during mvn install execution? What do you think?

**Comment by timothyjward** [comment]
> Why does the problem only emerge when one runs the tests in the same maven build and the same tests go just fine when you run them separately?

After I moved to Felix 5.6.1 as a framework the second test reliably failed every time, even when run separately.

**Comment by mrulli** [comment]
Ok, thank you, I'll try to fix the ClassNotFoundException using Felix. I think this could be closed.

**Comment by mrulli** [comment]
As a side note, I fixed the ClassNotFoundException with the Felix framework and I managed to get back to the original problem: the `mvn install` run OK for the single projects but was failing on the parent pom. 

Thus I made some other tests and it turned out that to solve the problem it is enough to **disable gogo shell**: as soon as I eliminate the gogo bundles from runrequires I manage to execute `mvn install` on the parent pom without any problem. 

I'll try to reproduce this behaviour in a simpler project than this to better isolate the real issue. Thank you again for your help.
